c++/src/kj/compat/http.c++

```diff
@@ -6402,8 +6726,16 @@ private:
 
     kj::Maybe<CompressionParameters> acceptedParameters;
     kj::String agreedParameters;
-    if (server.settings.enableWebSocketCompression) {
-      KJ_IF_MAYBE(value, requestHeaders.get(HttpHeaderId::SEC_WEBSOCKET_EXTENSIONS)) {
+    if (server.settings.webSocketCompressionMode != HttpServerSettings::NO_COMPRESSION) {
+      auto mode = server.settings.webSocketCompressionMode;
+      auto& target = mode == HttpServerSettings::AUTOMATIC_COMPRESSION ?
+          requestHeaders : (headers.isWebSocket() ? headers : requestHeaders);
```

Comment:
@kentonv curious what you think of this. I think we can determine if we're proxying by checking if `headers.isWebSocket()` is true (when we're in `MANUAL_COMPRESSION` mode). I ran some tests against EW and found that when acting as a server, `headers.isWebSocket() == false`, and when proxying, `headers.isWebSocket() == true`. Would this be preferable to:
> Hmm yeah that's a problem. It seems like we really want to specify this when we call acceptWebSocket(), at which point we know if we're proxying or not. The answer to this might be to set manual mode, but also have KJ provide a separate function that can be called to ask it to choose the appropriate response to a given extension request. Then the workers runtime will call that to set the header in the case that the WebSocket is being terminated locally.

Here's the actual fixup: https://github.com/capnproto/capnproto/pull/1505/commits/a6d42c2c4af81e4394bfa652e78c15ae7df2549d, I just commented on file so it would show up in the PR.

Reply:
I don't think this is a reliable way to detect proxying. I also don't think we should be detecting proxying. The point here is that some applications want to dictate exactly what compression options are used and some don't. Proxying is just a common reason why an app might want control, but it's entirely possible that a proxy might _want_ to add/remove compression, and that a non-proxy might want to avoid it for other reasons.

Reply:
Gotcha, I've removed this fixup and force pushed.